docs/EVE-IMAGE-SOURCES.md

```diff
@@ -0,0 +1,616 @@
+     image: lfedge/eve-watchdog:d4bf7ed4e3fa170061b8e6af6bd09ef7546c9c60-amd64
+     cgroupsPath: /eve/services/watchdog
+     oomScoreAdj: -1000
+   - name: xen-tools
+     image: lfedge/eve-xen-tools:1335e0e2d14d82ce3030f8d235f916474ae33bf4-amd64
+     cgroupsPath: /eve/services/xen-tools
+     oomScoreAdj: -999
+files:
+   - path: /etc/eve-release
+     contents: '0.0.0-master-12a6fb8d-kvm-amd64'     
+   - path: /etc/linuxkit-eve-config.yml
+     metadata: yaml
+```
+
+The above shows each container image used in creating this EVE bootable OS image. For example, the version of pillar used is
+`lfedge/eve-pillar:17837a9fcd05c765e9a1f6707b2e48f0f1dd215b-amd64`.
+
+The source for the builder configuration is rendered at build time from a series of templates.
+The base templates are in [images/](../images/). Specifically, the primary configuration file is the template
+[rootfs.yml.in](../images/rootfs.yml.in), which is rendered using templating, and potentially patched using
+other files in the same directory.
+
+As described above, when building an EVE image, the build process takes the final rendered configuration
+actually used and save it in the EVE image at `/etc/linuxkit-eve-config.yml`.
+
+## Container Images to Sources
+
+With the specific name and tag of each OCI image in hand, you can trace the source for each such package.
+
+The OCI container image can be pulled from Docker Hub using `docker pull`, for example:
+
+```sh
+docker pull lfedge/eve-pillar:17837a9fcd05c765e9a1f6707b2e48f0f1dd215b-amd64
+```
+
+and it can be inspected using any image inspection tool, including `docker image inspect`:
+
+```sh
+docker image inspect lfedge/eve-pillar:17837a9fcd05c765e9a1f6707b2e48f0f1dd215b-amd64
+```
+
+The results of the inspection will yield information about the container image, including tags that tell which git commit
+was used when building this image. In the above example:
+
+```json
+      "Labels": {
+        "org.mobyproject.config": "{\"capabilities\":[\"all\"],\"binds\":[\"/lib/modules:/lib/modules\",\"/dev:/dev\",\"/etc/resolv.conf:/etc/resolv.conf\",\"/r
+un:/run\",\"/config:/config\",\"/:/hostfs\",\"/persist:/persist:rshared,rbind\",\"/usr/bin/containerd:/usr/bin/containerd\"],\"devices\":[{\"path\":\"all\",\"ty
+pe\":\"a\",\"major\":0,\"minor\":0}],\"net\":\"host\",\"pid\":\"host\",\"rootfsPropagation\":\"shared\"}",
+        "org.mobyproject.linuxkit.revision": "unknown",
+        "org.mobyproject.linuxkit.version": "unknown",
+        "org.opencontainers.image.revision": "12a6fb8d29a6b47f998cd077dfb92213f0e6a55f",
+        "org.opencontainers.image.source": "https://github.com/linuxkit/linuxkit"
+      },
+```
+
+The git commit for the above source is given in the label `"org.opencontainers.image.revision": "12a6fb8d29a6b47f998cd077dfb92213f0e6a55f"`, i.e.
+you can get the same commit via `git checkout 12a6fb8d29a6b47f998cd077dfb92213f0e6a55f`.
+
+The above _should_ be the same as the current git commit.
```

Comment:
What does the _should_ imply? And why the reference to "current"? Did you mean what was current when it was built?

Reply:
No, I meant, the one that currently was checked out in the earlier line. I will clarify it.

Reply:
Updated, look when you have a chance, please.